.github/workflows/checks.yml

```diff
@@ -24,29 +21,28 @@ jobs:
         run: |
           mkdir -p /root/.ssh
           ssh-keyscan -t rsa github.com >> /root/.ssh/known_hosts
-
-      - name: Ignore Plaform Reqs if on PHP8
-        if: ${{ matrix.php == '8.0' }}
-        run: |
-          echo "COMPOSER_ARGS=--ignore-platform-reqs" >> $GITHUB_ENV
-          composer self-update --2
+          echo "COMPOSER_CACHE=$(composer config cache-dir)" >> $GITHUB_ENV
 
       - name: Composer Cache
         uses: actions/cache@v2
         with:
-          path: $(composer config cache-files-dir)
-          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
-          restore-keys: |
-            ${{ runner.os }}-composer-
+          path: |
+            ${{ env.COMPOSER_CACHE }}
+            vendor
+          key: ${{ runner.os }}-composer
 
       - name: Install dependencies
         run: |
-          composer install --no-progress --no-suggest --no-interaction ${{ env.COMPOSER_ARGS }} > /dev/null
+          composer update --no-progress --no-interaction > /dev/null
+
+      - name: PHPCS
+        if: matrix.php != '8.0'
```

Comment:
Do you think there's a way we can avoid this? Maybe replace the `@param`s with `@phpstan-param`s? Or maybe PHPCS can be told which PHP version to use?